### PR TITLE
Hc 276

### PR DESCRIPTION
**Pull request by DustinKLo**
https://hysds-core.atlassian.net/browse/HC-276

### CODE REVIEW

File: hysds_commons/job_utils.py [hunk 1]
```diff
@@ -148,8 +148,12 @@ def get_inputs(param, kwargs, rule=None, product=None):
         # If we are processing a list of products, create a list for outputs
         ret = process_xpath(source.split(":")[1], product)
 
+    # if the job param is optional but the value is not set (None/null)
+    if param.get('optional', False) and ret is None:
```
Comment (mcayanan) [inline hysds_commons/job_utils.py:152]: Should we check to see if there was a default set in hysds-io before setting this to empty? I know that from the UI, it populates with default automatically. But something like the trigger rules or if someone does a job submission via a script, seems like we would need a mechanism to populate that default value in here...

**Review by mcayanan — CHANGES_REQUESTED** [review]

File: hysds_commons/job_utils.py (hunk 1 above)
Comment (mcayanan) [reply hysds_commons/job_utils.py:152]: Here's an example of a job_submission script: https://github.jpl.nasa.gov/IEMS-SDS/nisar-pcm/blob/develop/conf/sds/files/test/pge/submit_pge_job.py

Comment (DustinKLo) [reply hysds_commons/job_utils.py:152]: good point ill add that change

Comment (DustinKLo) [reply hysds_commons/job_utils.py:152]: @mcayanan  actually i think the function already takes that into consideration
```
def get_inputs(param, kwargs, rule=None, product=None):
    """
    Update parameter to add in a value for the param
    @param param - parameter to update (param field in hysds-io)
    @param kwargs - job param inputs from user form
    @param rule - (optional) rule hit to use to fill pass through's
    @param product - (optional) product hit for augmenting
    """
    if "value" in param:  # Break out if value is known
        ret = param["value"]
        return ret

    source = param.get("from", "unknown")
    ret = param.get("default_value", None)  # Get a value  <------------ HERE

    if source == "submitter":
        ret = kwargs.get(param.get("name", "unknown"), None)
    elif source == "passthrough" and not rule is None:
        ret = rule.get(param["name"], None)
    elif source.startswith("dataset_jpath:") and not product is None:
        # If we are processing a list of products, create a list for outputs
        ret = process_xpath(source.split(":")[1], product)

    # if the job param is optional but the value is not set (None/null)
    if param.get('optional', False) and ret is None:
        return ""

    # Check value is found
    if ret is None and not (product is None) and not (rule is None):
        raise RuntimeError("Failed to find '{0}' input from '{1}'".format(param.get("name", "unknown"), source))
    return ret
```

Comment (mcayanan) [reply hysds_commons/job_utils.py:152]: ahhh. didn't see that. good catch.

**Review by mcayanan — APPROVED** [review]
LGTM.

**Pull request merged**